File: app/src/main/java/com/robotsandpencils/androiddaggerpracticemvp/lobby/LobbyGreetingContract.java
package com.robotsandpencils.androiddaggerpracticemvp.lobby;

/**
 * Created by pwray on 2017-11-01.
 */

interface LobbyGreetingContract {
    interface LobbyView {
        // button events
        void onCommonGreetingButtonClicked();
        void onLobbyGreetingButtonClicked();

        // greeting text actions
        void displayGreeting(String greeting);
        void hideGreeting();

        // greeting error actions
        void displayGreetingError(Throwable throwable);

        // loading indicator actions
        void setLoadingIndicator(boolean active);
    }

    interface LobbyPresenter {
        void loadCommonGreeting();
        void loadLobbyGreeting();
    }
}

File: app/src/main/java/com/robotsandpencils/androiddaggerpracticemvp/lobby/LobbyGreetingContract.kt
package com.robotsandpencils.androiddaggerpracticemvp.lobby

/**
 * Created by pwray on 2017-11-01.
 */

internal interface LobbyGreetingContract {
    interface LobbyView {
        // button events
        fun onCommonGreetingButtonClicked()

        fun onLobbyGreetingButtonClicked()

        // greeting text actions
        fun displayGreeting(greeting: String)

        fun hideGreeting()

        // greeting error actions
        fun displayGreetingError(throwable: Throwable)

        // loading indicator actions
        fun setLoadingIndicator(active: Boolean)
    }

    interface LobbyPresenter {
        fun loadCommonGreeting()
        fun loadLobbyGreeting()
    }
}
